Fixing `start_transaction`

szokeasaurusrex opened the pull request:
This is branched off of `ivana/potel/start-span` and rebased on latest `potel-base`. I added commit 971642baa382f5ff673a2b6fb51a370964f8e1fe to (hopefully) fix `start_transaction`.

sl0thentr0py commented on sentry_sdk/api.py:
```diff
@@ -334,22 +337,33 @@ def flush(
+    global _potel_span_init_kwargs
+
+    if _potel_span_init_kwargs is None:
+        _potel_span_init_kwargs = inspect.signature(tracing.POTelSpan).parameters.keys()
```
heavily against this kind of code, sorry
please write stupid code and list all the args, no introspection is necessary

sentrivana replied:
This kind of obfuscates what's going on -- I'd also prefer us to simply be stupidly explicit here since it makes it clearer what is happening.

szokeasaurusrex replied:
Fair enough, my reasoning was to avoid needing to update the arguments list in two places if it ever changes. But this way, it is perhaps less clear what is going on, so I will make this more explicit, as suggested.

szokeasaurusrex commented on sentry_sdk/tracing.py:
```diff
@@ -1285,10 +1283,19 @@ def __enter__(self):
             # set as the implicit current context
             self._ctx_token = context.attach(ctx)
 
+        self._scope_manager = sentry_sdk.new_scope()
+        scope = self._scope_manager.__enter__()
+        scope.span = self
+
         return self
 
     def __exit__(self, ty, value, tb):
         # type: (Optional[Any], Optional[Any], Optional[Any]) -> None
+        self._scope_manager.__exit__(ty, value, tb)
+        del (
+            self._scope_manager
+        )  # Eliminate circular reference which could prevent garbage collection
+
```
I added this to be able to access the current span. Is this code needed? Should we consider it in a separate PR, or is it okay to include here?

sl0thentr0py replied:
no I will take care of all scope related stuff, please remove both this and the above `new_scope`

szokeasaurusrex replied:
Done!

szokeasaurusrex commented:
@sl0thentr0py and @sentrivana, I addressed the PR comments, and rebased the PR on `ivana/potel/start-span`, since that branch has been updated from `potel-base`

sl0thentr0py reviewed (approved):
thx
